src/tool_operate.c

```diff
@@ -184,6 +184,63 @@ static curl_off_t VmsSpecialSize(const char *name,
 }
 #endif /* __VMS */
 
+#if defined(HAVE_UTIME) || \
+    (defined(WIN32) && (CURL_SIZEOF_CURL_OFF_T >= 8))
+static void setfiletime(long filetime, const char *filename,
+                        FILE *error_stream)
+{
+/* Windows utime() may attempt to adjust our unix gmt 'filetime' by a daylight
+   saving time offset and since it's GMT that is bad behavior. When we have
+   access to a 64-bit type we can bypass utime and set the times directly. */
+#if defined(WIN32) && (CURL_SIZEOF_CURL_OFF_T >= 8)
+#if (CURL_SIZEOF_LONG >= 8)
```

Comment:
I also moved the `if(filetime >= 0)` into the function. (I didn't do the ` if(filetime < 0) return` because then the `HANDLE` declaration would again be not the first statement in the block.)